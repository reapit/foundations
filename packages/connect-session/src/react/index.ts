import React from 'react'
import { ReapitConnectHook, ReapitConnectSession } from '../types'
import { ReapitConnectBrowserSession } from '../browser'

export const useReapitConnect = (reapitConnectBrowserSession: ReapitConnectBrowserSession): ReapitConnectHook => {
  const [connectSession, setConnectSession] = React.useState<ReapitConnectSession | null>(null)

  React.useEffect(() => {
    const connectGetSession = async () => {
      const session = await reapitConnectBrowserSession.connectSession()
      if (session) {
        setConnectSession(session)
      }
    }

    connectGetSession()
  }, [])

  const connectAuthorizeRedirect = React.useCallback(() => {
    reapitConnectBrowserSession.connectAuthorizeRedirect()
  }, [])

  const connectLoginRedirect = React.useCallback(() => {
    reapitConnectBrowserSession.connectLoginRedirect()
  }, [])

  const connectLogoutRedirect = React.useCallback(() => {
    reapitConnectBrowserSession.connectLogoutRedirect()
  }, [])

  const connectIsDesktop = reapitConnectBrowserSession.connectIsDesktop

  const connectHasSession = reapitConnectBrowserSession.connectHasSession

  return {
    connectSession,
    connectAuthorizeRedirect,
    connectLoginRedirect,
    connectLogoutRedirect,
    connectIsDesktop,
    connectHasSession,
  }
}
