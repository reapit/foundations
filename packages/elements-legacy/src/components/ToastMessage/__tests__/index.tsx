import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { ToastMessage, ToastMessageProps } from '..'

const baseProps: ToastMessageProps = {
  message: 'Toast message',
  variant: 'primary',
  onCloseToast: jest.fn(),
}
const defaultProps: ToastMessageProps = {
  ...baseProps,
  visible: true,
  displayDuration: 5000,
}

describe('ToastMessage', () => {
  it('should match a snapshot with default baseProps', () => {
    expect(toJson(shallow(<ToastMessage {...baseProps} />))).toMatchSnapshot()
  })

  it('should match a snapshot when pass different props', () => {
    expect(toJson(shallow(<ToastMessage {...defaultProps} />))).toMatchSnapshot()
  })

  it('should dismiss toast when click', () => {
    shallow(<ToastMessage {...defaultProps} visible={true} />)
      .find('[data-test="toast-wrapper"]')
      .first()
      .simulate('click')

    expect(defaultProps.onCloseToast).toHaveBeenCalledTimes(1)
  })

  afterEach(() => {
    jest.resetAllMocks()
  })
})
