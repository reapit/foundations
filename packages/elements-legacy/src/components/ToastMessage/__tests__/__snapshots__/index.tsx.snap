// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToastMessage should match a snapshot when pass different props 1`] = `
<div
  className="toast visible"
  data-test="toast-wrapper"
  onClick={[MockFunction]}
>
  <Button
    fullWidth={true}
    type="reset"
    variant="primary"
  >
    Toast message
  </Button>
</div>
`;

exports[`ToastMessage should match a snapshot with default baseProps 1`] = `
<div
  className="toast"
  data-test="toast-wrapper"
  onClick={[MockFunction]}
>
  <Button
    fullWidth={true}
    type="reset"
    variant="primary"
  >
    Toast message
  </Button>
</div>
`;
